### ISSUE TITLE
RUCSS - Incorrect notification displayed when RUCSS isn't created

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4126]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅3.9.0.4
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

On a customer's website, the pages that RUCSS picks to scan were not fetched at all.
Yet after the 1.5hr mark passed we displayed the following message:
`Remove Unused CSS is complete!`

while it was clearly not completed and nothing changed in the front-end: 

![](https://i.vgy.me/0IIPQY.jpg)

**To Reproduce**

I don't have a way to do this.

**Expected behavior**

A relevant notification should be displayed so the customer and support will know what the issue is.

**Additional context**

**Related ticket:** https://secure.helpscout.net/conversation/1547161846/272643?folderId=273766
Related to #4122. 

Maybe this should be tackled along with #3960.

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by mostafa-hisham** [comment]
**Reproduce the problem**
No, but i can see in the screenshots that  scan resources didn't fetch the resources

**Identify the root cause**
think we need to check if the pre warmup process was forced to stopped (stopped after 1houer)
which means that it failed
then shows a message that it failed 

**Scope a solution**
1- if 1 hour is passed we need to add a flag that it was force_stop in checker.php
https://github.com/wp-media/wp-rocket/blob/7bc0dd8dcc0ef5e2af156d978c8a437ec8bb1bd1/inc/Engine/Optimization/RUCSS/Warmup/Status/Checker.php#L71-L85
after line 82 set force_stop flag to true
`$this->set_warmup_force_stop();`

and add this function to set the flag true

```
	private function set_warmup_force_stop() {
		$prewarmup_stats                       = $this->options_api->get( 'prewarmup_stats', [] );
		$prewarmup_stats['force_stop'] = true;
		$this->options_api->set( 'prewarmup_stats', $prewarmup_stats );
	}
```
2- add function to get force_stop in RESTWP class  from status option
```
	private function get_force_stop(): bool {
		$prewarmup_stats = $this->options_api->get( 'prewarmup_stats', [] );
		return (bool) $prewarmup_stats['force_stop'];
	}
```

3- in respond_status we need to check if  warmup was force_stop

https://github.com/wp-media/wp-rocket/blob/7bc0dd8dcc0ef5e2af156d978c8a437ec8bb1bd1/inc/Engine/Optimization/RUCSS/Warmup/Status/RESTWP.php#L97-L123

```
		$allow_optimization = $this->get_allow_optimization();

		if ( $this->get_force_stop() ){
			$allow_optimization = false;
			return rest_ensure_response(
				$this->return_error( __( 'Pre-Warmup process did not finished.', 'rocket' ),
					[
						'allow_optimization' => $allow_optimization,
						'force_stop' => $this->get_force_stop(),
					]
				)
			);
		}
```

4- at the start of warmup set the flag to false
`'force_stop'      => false,`
https://github.com/wp-media/wp-rocket/blob/7bc0dd8dcc0ef5e2af156d978c8a437ec8bb1bd1/inc/Engine/Optimization/RUCSS/Warmup/Scanner.php#L133-L137

**NOTE** **this may need edit in react component to change UI**

**Estimate the effort**
[S]

**Comment by remyperona** [comment]
@wp-media/productrocket During the alpha, with the first iteration of the pre-warmup, we were showing error messages and files that were not warmed-up. We decided together with the product & support team to not display this information, and just move forward and enable RUCSS after 1 hour.

The information is still available when enabling the `WP_ROCKET_RUCSS_DEBUG` constant too.

This issue is asking us to roll back to that. Do we want to go forward with this?

**Comment by GeekPress** [comment]
> A relevant notification should be displayed so the customer and support will know what the issue is.

@vmanthos Could you explain to which kind of « relevant notification » you are referring to with an example of what you would like to see?

**Comment by vmanthos** [comment]
I apologize for missing the notification for this one.

When no resources are fetched, RUCSS is not going to work as expected. Until now, I've seen this happening when a plugin, e.g. Wordfence, or a server's security, e.g. `ModSecurity` on `O2switch`, is blocking WP Rocket's fetching requests.

When:
1. Pages have been found and scanned `AND`
2. No resources have been fetched during the one hour we wait to activate the feature,

we can:
1. do not automatically enable the feature and 
2.  display a message like this:
```
The Remove Unused CSS feature was not activated because the process could not find CSS and JavaScript files on the pages it scanned. To resolve this please follow this guide.
```
"**this guide**" will reference a doc where we can gather the common causes of the issue we discover and the respective solutions.

**Comment by GeekPress** [comment]
@camilamadronero Can we plan to work on the "guide" mentioned 👆

**Comment by camilamadronero-zz** [comment]
Sure thing! Asana card created here: https://app.asana.com/0/1200521205873018/1201354770665024.

**Comment by piotrbak** [comment]
Closed by 3.11. The warmup is not used anymore.
